com.musala.xml.parser/src/com/musala/xml/parser/Main.java

```diff
@@ -0,0 +1,132 @@
+package com.musala.xml.parser;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+
+import com.musala.xml.parser.model.School;
+import com.musala.xml.parser.model.Student;
+import com.musala.xml.parser.model.Teacher;
+
+public class Main {
+
+    public static void main(String[] args) throws IOException, JAXBException {
+
+        ClassLoader classload = Main.class.getClassLoader();
+        File configFile = new File(classload.getResource(Constants.XML_INTERNAL_FILE_PATH).getFile());
+        JAXBContext jaxb = JAXBContext.newInstance(School.class);
+        Unmarshaller jaxbUnmarsheller = jaxb.createUnmarshaller();
+        School mySchool = (School) jaxbUnmarsheller.unmarshal(configFile);
+        Print(mySchool);
+    }
+
+    private static void Print(School myschool) {
+
+        int numberOfClasses = myschool.getSchoolClasses().size();
+        System.out.printf("The school name is: %s!%n There %s %d school classes.%n", myschool.getName(),
+                numberOfClasses > 1 ? "are" : "is", numberOfClasses);
+        printClassesData(myschool, numberOfClasses);
+
+    }
+
+    private static void printClassesData(School myschool, int numberOfClasses) {
+        for (int currentClassNum = 0; currentClassNum < numberOfClasses; currentClassNum++) {
+            printStudents(myschool, currentClassNum);
+            printTeacher(myschool, currentClassNum);
+        }
+    }
+
+    private static void printStudents(School myschool, int currentClassNum) {
+        String classNum = turnClassNumberIntoText(currentClassNum).toString();
+        System.out.printf("The %s class have %d students: %n", classNum,
+                myschool.getSchoolClasses().get(currentClassNum).getStudentsReference().size());
+        for (int j = 0; j < myschool.getSchoolClasses().get(currentClassNum).getStudentsReference().size(); j++) {
+            String currentReference = myschool.getSchoolClasses().get(currentClassNum).getStudentsReference().get(j);
+            Student currentStudent = (Student) searchObjectByReference(currentReference, myschool, "Student");
+            System.out.printf("%s is %d years old with fac. number %s.%n", currentStudent.getName(),
+                    currentStudent.getAge(), currentStudent.getFacultyNumber());
+        }
+    }
+
+    private static void printTeacher(School myschool, int currentClassNum) {
+        String currentTeacherReference = myschool.getSchoolClasses().get(currentClassNum).getTeacherReference();
```

Comment:
No need to declare `currentTeacherReference` and then pass it to the `searchObjectByReference()` method. You can directly pass `myschool.getSchoolClasses().get(currentClassNum).getTeacherReference()` as the first argument of `searchObjectByReference()` and apply formatter - readability will not suffer whatsoever.